enhancements/windows-containers/windows-machine-config-operator.md

```diff
@@ -0,0 +1,165 @@
+(TBD). Once the Machine object is in the "provisioned" state, the operator will
+connect to it transfer the binaries and perform the following steps for it to
+join the cluster as a worker node:
+* Transfer the required binaries
+* Configure the kubelet service using WMCB
+* Run the hybrid-overlay-node.exe binary
+* Configure the kubelet service for CNI using WMCB
+* Configure kube-proxy
+
+The operator expects the cluster admin to create a well know Secret in it's
+namespace with a private key that will be used to interact with the Windows
+instance. The operator will check for this secret during boot up time and create
+a user data Secret which the cluster admin can reference in the Windows
+MachineSet that they create.
+
+TBD: How do we get MCO to ignore Windows nodes?
+
+### Risks and Mitigations
+
+The main risk is that OVN Hybrid networking will not GA in 4.6. In that scenario
+an alternative networking solution like Calico will need to be explored.
+
+### Test Plan
+
+The WMCO repositories has already been integrated with Prow CI. We are running
+e2e tests for every PR that is opened. These e2e tests involve bringing up a
+cluster on all supported cloud providers, instantiating Windows nodes and
+running the following tests:
+* Basic node validation
+* East west networking between Linux and Windows workloads
+* North south networking
+
+### Graduation Criteria
+
+This enhancement will start as GA.
+
+### Upgrade / Downgrade Strategy
+
+This is an involved topic and we will be opening a separate enhancement
+detailing it.
+
+### Version Skew Strategy
+
+We plan to maintain kubelet major version parity with the Linux counterpart.
+
+
+## Implementation History
+
+We already have a working [prototype](https://github.com/openshift/windows-machine-config-bootstrapper)
+in place where use a custom [CRD](https://github.com/openshift/windows-machine-config-operator/blob/master/deploy/crds/wmc.openshift.io_windowsmachineconfigs_crd.yaml)
+and the [Windows Node Installer](https://github.com/openshift/windows-machine-config-bootstrapper/tree/master/tools/windows-node-installer)
+library to enable Windows workloads.
+
+## Drawbacks
+
+The Microsoft container ecosystem is not fully mature or on par with Linux
+containers. For one we can only support Windows process containers that
+restricts workloads that are runnable on Windows Server 2019 as there is a tight
+kernel version coupling with Windows containers. The coupling is slightly
+loosened if we use Hyper-V containers but that is not a viable option yet with
+all cloud providers as it requires nested virtualization support. Given these
+limitations customers could potentially get a degraded experience with Windows
+workloads when compared to Linux workloads on the cluster.
+
+## Alternatives
+
+Instead of using the MachineSet as the point of entry to enable Windows
+workloads, we can introduce a Windows specific CRD to do the same as we have
+done in the prototype.
```

Comment:
Should we add link to the CRD ?

Reply:
I have linked to the CRD in the implementation history.